packages/portfolio-contract/src/portfolio.flows.ts

```diff
@@ -1,30 +1,114 @@
+import type { GuestInterface } from '@agoric/async-flow';
+import { Any } from '@agoric/cosmic-proto/google/protobuf/any.js';
+import { MsgLock } from '@agoric/cosmic-proto/noble/dollar/vaults/v1/tx.js';
+import { MsgSwap } from '@agoric/cosmic-proto/noble/swap/v1/tx.js';
+import { makeTracer } from '@agoric/internal';
 import type {
+  CosmosChainAddress,
   OrchestrationAccount,
   OrchestrationFlow,
   Orchestrator,
 } from '@agoric/orchestration';
-import type { TargetApp } from '@agoric/vats/src/bridge-target.js';
-import type { Passable } from '@endo/pass-style';
+import { coerceAccountId } from '@agoric/orchestration/src/utils/address.js';
+import type { ZoeTools } from '@agoric/orchestration/src/utils/zoe-tools.js';
+import type { ZCFSeat } from '@agoric/zoe';
+import type { ResolvedPublicTopic } from '@agoric/zoe/src/contractSupport/topics.js';
+import type { PortfolioKit } from './portfolio.exo.ts';
+import type { ProposalShapes } from './type-guards.ts';
 
-export const makeHookAccount = (async (
+const trace = makeTracer('PortF');
+
+export const makeLocalAccount = (async (orch: Orchestrator, _ctx: unknown) => {
+  const agoricChain = await orch.getChain('agoric');
+  const account = (await agoricChain.makeAccount()) as OrchestrationAccount<{
+    chainId: 'agoric-any';
+  }>;
+
+  return account;
+}) satisfies OrchestrationFlow;
+harden(makeLocalAccount);
+
+// TODO: push down to NobleMethods
+const makeSwapLockMessages = (
+  nobleAddr: CosmosChainAddress,
+  amount: Amount<'nat'>,
+) => {
+  const [poolId, denom, denomTo] = [0n, 'uusdc', 'uusdn'];
+  const msgSwap: MsgSwap = {
+    signer: nobleAddr.value,
+    amount: { denom, amount: `${amount.value}` },
+    routes: [{ poolId, denomTo }],
+    // TODO: swap min multiplier?
```

Comment:
_low priority_

Offer both `amount` in USDC and `ratio` that converts it to USDN

Reply:
That sounds good!